pkg/bpf/bpf.go

```diff
@@ -305,6 +314,10 @@ func OpenOrCreateMap(path string, mapType int, keySize, valueSize, maxEntries ui
 		if err != nil {
 			return 0, isNewMap, err
 		}
+
+		// Need to close this fd since the next ObjGet
+		// returns a different fd.
+		ObjClose(fd)
```

Comment:
Yes, that one should still be leaked in ObjPin()'s error path. I don't see where we close the created map in such case.